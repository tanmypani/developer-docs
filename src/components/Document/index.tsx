import React, { useEffect } from 'react';
import './index.scss';
import { customizeDocContent, addScrollListener } from './helper';
import Footer from '../Footer';
import Breadcrums from '../Breadcrums';
import LinkableHeader from '../LinkableHeader';
import { HOME_PAGE_ID } from '../../configs/doc-configs';
import parse, { HTMLReactParserOptions, domToReact, attributesToProps } from 'html-react-parser';

const Document = (props: {
    pageid?: string;
    docTitle: string;
    docContent: string;
    isPublicSiteOpen: boolean;
    shouldShowRightNav: boolean
    breadcrumsData: any;
}) => {
    useEffect(() => {
        customizeDocContent();
    }, [props.docContent]);

    useEffect(() => {
        addScrollListener();
    }, []);

    const options: HTMLReactParserOptions = {
        replace: (domNode: any) => {
            if (domNode.type === 'tag' &&
                ['h2', 'h3', 'h4'].includes(domNode.name) &&
                !domNode.parent?.attribs?.class?.includes('non-link')
            ) {
                const props = attributesToProps(domNode.attribs);
                return (<LinkableHeader {...props} tag={domNode.name} id={domNode.attribs.id}>
                    {domToReact(domNode.children, options)}
                </LinkableHeader>)
            }
        }
    };

    return (
        <div
            className="documentWrapper"
            style={{
                width: !props.shouldShowRightNav ? '100%' : null,
            }}
        >
            {props.pageid !== HOME_PAGE_ID && (
                <Breadcrums
                    breadcrumsData={props.breadcrumsData}
                    pageid={props.pageid}
                />
            )}
            <div
                id={props.docTitle}
                className="documentView"
            >
                {parse(props.docContent, options)}
            </div>
            {props.isPublicSiteOpen && <Footer />}
        </div>
    );
};

export default Document;
