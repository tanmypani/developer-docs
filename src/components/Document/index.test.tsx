import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Document from './index';
import { addScrollListener, customizeDocContent } from './helper';


jest.mock('../../utils/lang-utils', () => (key) => key);

jest.mock('../Footer', () => () => <div data-testid="footerDiv" />);

jest.mock('./helper', () => ({
  customizeDocContent:  jest.fn(),
  addScrollListener:  jest.fn(),
}));

describe('Document', () => {
const docTitle = 'test title';
const docContent = 'test content';
const isPublicSiteOpen = false;

    it('should render correctly if isPublicSiteOpen is false', () => {
        const { container } = render(<Document docTitle={docTitle} docContent={docContent} isPublicSiteOpen={isPublicSiteOpen} />);
      expect(container).toMatchSnapshot();
    })
    it('should render correctly if isPublicSiteOpen is true', () => {
        const { container, getByTestId } = render(<Document docTitle={docTitle} docContent={docContent} isPublicSiteOpen={!isPublicSiteOpen} />);
        fireEvent.scroll(window, { target: { scrollY: 100 } })
        expect(addScrollListener).toHaveBeenCalledTimes(1);
        expect(customizeDocContent).toHaveBeenCalledTimes(1);
     
      expect(getByTestId('footerDiv')).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    })
})