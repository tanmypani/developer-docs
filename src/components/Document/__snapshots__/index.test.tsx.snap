// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document should render correctly if isPublicSiteOpen is false 1`] = `
<div>
  <div
    class="documentWrapper"
    style="width: 100%;"
  >
    <div
      class="documentView"
      id="test title"
    >
      test content
    </div>
  </div>
</div>
`;

exports[`Document should render correctly if isPublicSiteOpen is true 1`] = `
<div>
  <div
    class="documentWrapper"
    style="width: 100%;"
  >
    <div
      class="documentView"
      id="test title"
    >
      test content
    </div>
    <div
      data-testid="footerDiv"
    />
  </div>
</div>
`;
